File: SubwayArrivalFlutterApp/lib/services/subway_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../models/arrival_info.dart';
import 'api_key.dart';

class SubwayService {

  static Future<List<ArrivalInfo>> fetchArrivalInfo(
    String station, {
    int startIndex = 0,
    int endIndex = 5,
    String type = 'json',
    String service = 'realtimeStationArrival',
  }) async {
    try {
      final encoded = Uri.encodeComponent(station);
      final url = Uri.parse(
          'https://swopenAPI.seoul.go.kr/api/subway/$subwayApiKey/$type/$service/$startIndex/$endIndex/$encoded');

      final response = await http.get(url);
      if (response.statusCode == 200) {
        final json = jsonDecode(response.body) as Map<String, dynamic>;
        final List list = json['realtimeArrivalList'] as List;
        return list.map((e) => ArrivalInfo.fromJson(e)).toList();
      } else {
        throw Exception('HTTP ${response.statusCode}');
      }
    } catch (e) {
      print('Error fetching arrival info: $e');
      return [];
    }
  }
}

File: SubwayArrivalApp/Services/SubwayService.swift
import Foundation

class SubwayService {
    static let shared = SubwayService()
    private init() {}

    private let apiKey = "YOUR_API_KEY" // API 키를 여기에 입력하세요

    func fetchArrivalInfo(for station: String, completion: @escaping ([ArrivalInfo]) -> Void) {
        let encoded = station.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) ?? station
        let urlString = "https://swopenAPI.seoul.go.kr/api/subway/\(apiKey)/json/realtimeStationArrival/0/2/\(encoded)"
        guard let url = URL(string: urlString) else {
            completion([])
            return
        }

        let task = URLSession.shared.dataTask(with: url) { data, response, error in
            guard let data = data else {
                completion([])
                return
            }
            do {
                let response = try JSONDecoder().decode(ArrivalResponse.self, from: data)
                let infos = response.realtimeArrivalList.map { ArrivalInfo.from($0) }
                completion(infos)
            } catch {
                completion([])
            }
        }
        task.resume()
    }
}
